lang/rust/Makefile

```diff
@@ -0,0 +1,105 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+VENV := precommit_venv
+HOOKS := .git/hooks/pre-commit
+
+# PRE-COMMIT HOOKS
+
+$(VENV): .requirements-precommit.txt
+	virtualenv -p python3 $(VENV)
+	$(VENV)/bin/pip install -r .requirements-precommit.txt
+
+.PHONY: env
+env: $(VENV)
+
+.PHONY: clean-env
+clean-env:
+	rm -rf $(VENV)
```

Comment:
This looks dangerous. Can we be sure $(VENV) expands legitimately and doesn't contain globs?